### PR TITLE
Upgrade maturin

### PR DESCRIPTION
**Pull request by mbway**
upgrade to latest packages.

(draft: waiting on https://github.com/PyO3/maturin-import-hook/pull/8 )

### CODE REVIEW

File: src/maturin_import_hook/settings.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def to_args(self) -> list[str]:
 class MaturinBuildSettings(MaturinSettings):
     """settings for `maturin build`."""
 
-    skip_auditwheel: bool = False
+    auditwheel: Optional[str] = None
```
Comment (mbway) [inline src/maturin_import_hook/settings.py:102]: technically this means the import hook doesn't have 100% compatibility with older maturin versions but I don't think it's worth (A) maintaining backwards compatibility or (B) bumping the minimum supported version

**Comment by mbway** [comment]
currently `maturin new --bindings` creates projects using `pyo3 == 0.22.0` which is why the tests are failing because I upgraded the projects to be `0.23.1`. https://github.com/PyO3/maturin/pull/2308 is merged and changes to use `0.23.1` but I think this PR should wait until the next maturin release so the import hook is tied to a release tag of maturin and not a random commit

File: src/maturin_import_hook/settings.py (hunk 1 above)
Comment (messense) [reply src/maturin_import_hook/settings.py:102]: I think it's fine.

**Comment by mbway** [comment]
need to wait for https://github.com/PyO3/maturin/issues/2347 and https://github.com/PyO3/maturin/issues/2346 to fix the remaining test failures.

**Comment by mbway** [comment]
tests now pass locally. Waiting for the next maturin release to pin to

**Review by messense — APPROVED** [review]

**Pull request merged**